File: Python/Calculadora_Basica_Funciones.py
def operaciones(opcion, num1, num2):
    if opcion == '1':
        resultado = num1 + num2
        return f"{num1} + {num2} es: {resultado}"
    elif opcion == '2':
        resultado = num1 - num2
        return f"{num1} - {num2} es: {resultado}"
    elif opcion == '3':
        resultado = num1 * num2
        return f"{num1} * {num2} es: {resultado}"
    elif opcion == '4':
        if num2 != 0:
            resultado = num1 / num2
            return f"{num1} / {num2} es: {resultado}"
        else:
            return "Error: División por cero no permitida."
    return "Error: Opción no válida."

def menu():
    print("\nSeleccione una operación:")
    print("\n1. Sumar")
    print("2. Restar")
    print("3. Multiplicar")
    print("4. Dividir")
    print("5. Salir")
    return input("\nIngrese el número de la operación que desea realizar: ")

def main():
    print("Bienvenido a la Calculadora Básica")
    while True:
        opcion = menu()
        if opcion == '5':
            print("\nGracias por usar la calculadora. ¡Hasta luego!")
            break
        elif opcion in ['1', '2', '3', '4']:
            try:
                num1 = float(input("\nIngrese el primer número: "))
                num2 = float(input("\nIngrese el segundo número: "))
                resultado_str = operaciones(opcion, num1, num2)
                print(f"\nEl resultado de {resultado_str}")
            except ValueError:
                print("Error: Por favor ingrese un número válido.")
        else:
            print("\nOpción no válida. Por favor seleccione una opción del 1 al 5.")

main()

File: aJava/src/Calculadora_Basica_Funciones.java
import java.util.Scanner;

public class Calculadora_Basica_Funciones {
    
    public static String operaciones(String opcion, double num1, double num2) {
        switch (opcion) {
            case "1" -> {
                double resultado = num1 + num2;
                return num1 + " + " + num2 + " es: " + resultado;
            }
            case "2" -> {
                double resultado = num1 - num2;
                return num1 + " - " + num2 + " es: " + resultado;
            }
            case "3" -> {
                double resultado = num1 * num2;
                return num1 + " * " + num2 + " es: " + resultado;
            }
            case "4" -> {
                if (num2 != 0) {
                    double resultado = num1 / num2;
                    return num1 + " / " + num2 + " es: " + resultado;
                } else {
                    return "Error: División por cero no permitida.";
                }
            }
            default -> {
            }
        }
        return "Error: Opción no válida.";
    }
    
    public static String menu(Scanner scanner) {
        System.out.println("\nSeleccione una operación:");
        System.out.println("\n1. Sumar");
        System.out.println("2. Restar");
        System.out.println("3. Multiplicar");
        System.out.println("4. Dividir");
        System.out.println("5. Salir");
        System.out.print("\nIngrese el número de la operación que desea realizar: ");
        return scanner.nextLine();
    }
    
    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.println("Bienvenido a la Calculadora Básica");
            OUTER:
            while (true) {
                String opcion = menu(scanner);
                switch (opcion) {
                    case "5" -> {
                        System.out.println("\nGracias por usar la calculadora. ¡Hasta luego!");
                        break OUTER;
                    }
                    case "1", "2", "3", "4" -> {
                        try {
                            System.out.print("\nIngrese el primer número: ");
                            double num1 = Double.parseDouble(scanner.nextLine());
                            System.out.print("\nIngrese el segundo número: ");
                            double num2 = Double.parseDouble(scanner.nextLine());
                            String resultadoStr = operaciones(opcion, num1, num2);
                            System.out.println("\nEl resultado de " + resultadoStr);
                        } catch (NumberFormatException e) {
                            System.out.println("Error: Por favor ingrese un número válido.");
                        }
                    }
                    default -> System.out.println("\nOpción no válida. Por favor seleccione una opción del 1 al 5.");
                }
            }
        }
    }
}